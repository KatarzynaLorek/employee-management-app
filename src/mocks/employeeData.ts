import { IResponseObject } from '../types/responses';

export const mockEmployeeData: IResponseObject = {
  id: '1',
  firstName: 'mockFirstName',
  lastName: 'mockLastname',
  position: 'junior',
  department: 'IT',
};
