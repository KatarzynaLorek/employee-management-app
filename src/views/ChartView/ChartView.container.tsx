import React, { useState } from 'react';
import './ChartView.scss';
import ChartView from './ChartView';
import { useAppSelector } from '../../hooks/reduxHooks';
import { IResponseObject } from '../../types/responses';
import { selectAll } from '../../store/reducers/employees';

const ChartViewContainer = (): JSX.Element => {
  const [data, setData] = useState<IResponseObject[]>([]);

  const employees = useAppSelector((state) => selectAll(state));

  setTimeout(() => setData(employees), 200);

  return <ChartView data={data} />;
};

export default ChartViewContainer;
