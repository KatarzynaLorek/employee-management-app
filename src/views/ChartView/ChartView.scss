.chartview {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: 100vh;

  &_navbar {
    display: flex;
    flex-direction: column;
    align-self: flex-start;
    width: 300px;
    padding: 200px 50px 0;
  }

  &_input {
    margin-bottom: 15px;
  }

  &_chartContainer {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 90vw;
    max-width: calc(100vw - 600px);
    height: 100%;
    padding: 100px 10px 100px 0;
  }
}

@media (max-width: 1250px) {
  .chartview {
    padding-top: 30px;

    &_chartContainer {
      align-self: flex-start;
      max-height: 800px;
      max-width: calc(100vw - 300px);
    }
  }
}
@media (max-width: 1000px) {
  .chartview {
    flex-direction: column;

    &_navbar {
      display: flex;
      flex-direction: row;
      justify-content: center;
      width: 100%;
      padding: 100px 0 0 0;
    }

    &_input {
      padding-left: 20px;
    }

    &_chartContainer {
      align-self: center;
      max-width: calc(100vw - 30px);
      padding: 30px 0px 30px 0;
    }
  }
}

@media (max-width: 768px) {
  .chartview {
    &_chartContainer {
      width: 100%;
      max-width: calc(100vw - 30px);
    }
  }
}

@media (max-width: 450px) {
  .chartview {
    &_navbar {
      display: flex;
      flex-direction: column;
      align-items: center;
    }

    &_input {
      padding: 0 20px;
      width: 100%;
    }
  }
}
