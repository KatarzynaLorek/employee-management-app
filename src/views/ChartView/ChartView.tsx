import React from 'react';

const ChartView: React.FC = () => <div>Chart view</div>;

export default ChartView;
