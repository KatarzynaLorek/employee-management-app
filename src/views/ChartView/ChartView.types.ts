import { IResponseObject } from '../../types/responses';

export interface IChartView {
  data: IResponseObject[];
}
