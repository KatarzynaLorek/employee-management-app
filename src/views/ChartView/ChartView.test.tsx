import React from 'react';
import { render } from '@testing-library/react';
import ChartView from './ChartView';
import { mockEmployeeData } from '../../mocks/employeeData';

jest.mock('react-chartjs-2', () => ({
  Bar: () => null,
  Doughnut: () => null,
}));

describe('ChartView component', () => {
  it('renders properly', () => {
    const container = render(<ChartView employeesData={[mockEmployeeData]} />);
    expect(container.baseElement).toMatchSnapshot();
  });
});
