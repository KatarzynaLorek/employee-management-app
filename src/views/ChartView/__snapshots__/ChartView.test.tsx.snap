// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartView component renders properly 1`] = `
<body>
  <div>
    <div
      class="chartview"
    >
      <div
        class="chartview_navbar"
      >
        <div
          class="chartview_input"
        >
          <label
            class="input_label"
            for="chartInput"
          >
            Chart Type
          </label>
          <select
            class="input_select"
            name="chartInput"
          >
            <option
              value="bar"
            >
              bar
            </option>
            <option
              value="doughnut"
            >
              doughnut
            </option>
          </select>
        </div>
        <div
          class="chartview_input"
        >
          <label
            class="input_label"
            for="chartInput"
          >
            Sort by
          </label>
          <select
            class="input_select"
            name="chartInput"
          >
            <option
              value="department"
            >
              department
            </option>
            <option
              value="position"
            >
              position
            </option>
          </select>
        </div>
      </div>
      <div
        class="chartview_chartContainer"
      />
    </div>
  </div>
</body>
`;
